src/mds/MDCache.h

```diff
@@ -129,6 +129,8 @@ class MDCache {
   using clock = ceph::coarse_mono_clock;
   using time = ceph::coarse_mono_time;
 
+  vector<inodeno_t> consistent_hash_inodes;
```

Comment:
This should be an `elist<CInode*>.

Reply:
You also don't remove inodes anywhere when they are dropped from the cache (migrated or dropped).

Reply:
I didn't exactly understand this. Do you mean that when the inodes are trimmed, then it would be difficult to track the dropped inodes (consistent hashed ones) using just a vector of inode nos alone? I did fix the migration case (updating the vector) but the trimming part is interesting. Looking into it.

Reply:
You only need to do random insert, random erase, and iteration. That indicates a list is appropriate. (No lookup and no need for duplicate detection! So set is unnecessary.)

`elist` is an efficient way to create a list of `CInode`s without the overhead of `std::list`. It's used extensively in the MDS, just grep for it in `CInode.h`.

Reply:
Ack. Still a bit unclear about this. Why do we need an elist of `CInodes` when all we need to track are the inode numbers?

Reply:
Because you need to do `get_inode` to fetch the `CInode` in the cache, which is extra overhead. Additionally, it makes it clear that anything in `consistent_hash_inodes` **must** be in the MDS cache. (The MDS should not care about pinned inodes on other ranks.)

Reply:
Thanks! That is a reasonable justification. I will change the data structure to `elist`.